src/lib/settings/GatewayDriver.js

```diff
@@ -92,13 +61,11 @@ class GatewayDriver {
 	register(name, { provider = this.client.options.providers.default, schema = new Schema() } = {}) {
 		if (typeof name !== 'string') throw new TypeError('You must pass a name for your new gateway and it must be a string.');
 		if (!(schema instanceof Schema)) throw new TypeError('Schema must be a valid Schema instance.');
-		if (this.name !== undefined && this.name !== null) throw new Error(`The key '${name}' is either taken by another Gateway or reserved for GatewayDriver's functionality.`);
+		if (this.has(name)) throw new Error(`The key '${name}' is either taken by another Gateway or reserved for GatewayDriver's functionality.`);
```

Comment:
Error is wrong, can only be taken by another gateway now that it'ss powered by map
```suggestion
		if (this.has(name)) throw new Error(`The key '${name}' is taken by another Gateway.`);
```